yt_dlp/networking/common.py

```diff
@@ -263,26 +269,25 @@ def _check_proxies(self, proxies):
             if scheme not in self._SUPPORTED_PROXY_SCHEMES:
                 raise UnsupportedRequest(f'Unsupported proxy type: "{scheme}"')
 
-    def _check_cookiejar_extension(self, extensions):
-        if not extensions.get('cookiejar'):
-            return
-        if not isinstance(extensions['cookiejar'], CookieJar):
-            raise UnsupportedRequest('cookiejar is not a CookieJar')
-
     def _check_timeout_extension(self, extensions):
-        if extensions.get('timeout') is None:
-            return
-        if not isinstance(extensions['timeout'], (float, int)):
-            raise UnsupportedRequest('timeout is not a float or int')
+        if not isinstance(extensions.pop('timeout', None), (float, int, type(None))):
+            raise TypeError('timeout extension is not a float or int')
+
+    def _check_cookiejar_extension(self, extensions):
+        if not isinstance(extensions.pop('cookiejar', None), (CookieJar, type(None))):
+            raise TypeError('cookiejar extension is not a CookieJar')
 
     def _check_extensions(self, extensions):
-        self._check_cookiejar_extension(extensions)
-        self._check_timeout_extension(extensions)
+        """Check extensions for unsupported extensions. To be implemented by subclasses."""
```

Comment:
I'd do this (+other comment) since type checking isn't really dependant on whether the RH actually supports it or not. But I won't insist.
```py
from types import NoneType

    def _check_extensions(self, extensions):
        """Check extensions for unsupported extensions. To be implemented by subclasses."""
        assert isinstance(extensions.get('cookiejar'), (CookieJar, NoneType))
        assert isinstance(extensions.get('timeout'), (float, int, NoneType))
```

Reply:
Or maybe this check should be done at Request creation?

Reply:
ah yeah true.

> Or maybe this check should be done at Request creation?
nah request doesn't know about the extensions